test/AutoDiff/differentiable_attr_type_checking.swift

```diff
@@ -540,10 +540,24 @@ struct JVPStruct {
   let p: Float
 }
 
+extension JVPStruct : VectorNumeric {
+  static var zero: JVPStruct { return JVPStruct(p: 0) }
+  static func + (lhs: JVPStruct, rhs: JVPStruct) -> JVPStruct {
+    return JVPStruct(p: lhs.p + rhs.p)
+  }
+  static func - (lhs: JVPStruct, rhs: JVPStruct) -> JVPStruct {
+    return JVPStruct(p: lhs.p - rhs.p)
+  }
+  typealias Scalar = Float
+  static func * (lhs: Float, rhs: JVPStruct) -> JVPStruct {
+    return JVPStruct(p: lhs * rhs.p)
+  }
+}
+
 extension JVPStruct : Differentiable {
   typealias TangentVector = JVPStruct
   typealias CotangentVector = JVPStruct
-  func moved(toward direction: JVPStruct) -> JVPStruct {
+  func moved(along direction: JVPStruct) -> JVPStruct {
```

Comment:
You can remove this and the next method. They provided by default when Tangent == Self

Reply:
Thanks! Done